bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/EpsonProjectorDevice.java

```diff
@@ -0,0 +1,621 @@
+            195, 207, 219, 231, 243 };
+
+    private static final int[] MAP18 = new int[] { 0, 13, 26, 40, 53, 67, 80, 94, 107, 121, 134, 148, 161, 175, 188,
+            202, 215, 229, 242 };
+
+    private static final int[] MAP_COLOR_TEMP = new int[] { 0, 25, 51, 76, 102, 128, 153, 179, 204, 230 };
+    private static final int[] MAP_FLESH_COLOR = new int[] { 0, 36, 73, 109, 146, 182, 219 };
+
+    private static final int DEFAULT_TIMEOUT = 5 * 1000;
+    private static final int POWER_ON_TIMEOUT = 100 * 1000;
+    private static final int POWER_OFF_TIMEOUT = 130 * 1000;
+    private static final int LAMP_REFRESH_WAIT_MINUTES = 5;
+
+    private static final String ON = "ON";
+    private static final String ERR = "ERR";
+
+    private Logger logger = LoggerFactory.getLogger(EpsonProjectorDevice.class);
+
+    @Nullable
+    private ScheduledExecutorService scheduler = null;
+    private EpsonProjectorConnector connection;
+    private ExpiringCache<Integer> cachedLampHours = new ExpiringCache<>(Duration.ofMinutes(LAMP_REFRESH_WAIT_MINUTES),
+            this::queryLamp);
+    private boolean connected = false;
+    private boolean ready = false;
+
+    public EpsonProjectorDevice(SerialPortManager serialPortManager, EpsonProjectorConfiguration config) {
+        connection = new EpsonProjectorSerialConnector(serialPortManager, config.serialPort);
+        ready = true;
+    }
+
+    public EpsonProjectorDevice(EpsonProjectorConfiguration config) {
+        connection = new EpsonProjectorTcpConnector(config.host, config.port);
+        ready = true;
+    }
+
+    public boolean isReady() {
+        return ready;
+    }
+
+    public void setScheduler(ScheduledExecutorService scheduler) {
+        this.scheduler = scheduler;
+    }
+
+    private synchronized @Nullable String sendQuery(String query, int timeout)
+            throws EpsonProjectorCommandException, EpsonProjectorException {
+        logger.debug("Query: '{}'", query);
+        String response = connection.sendMessage(query, timeout);
+
+        if (response.length() == 0) {
+            throw new EpsonProjectorException("No response received");
+        }
+
+        response = response.replace("\r:", "");
+        logger.debug("Response: '{}'", response);
+
+        if (ERR.equals(response)) {
+            throw new EpsonProjectorCommandException("Error response received for command: " + query);
+        }
+
+        if ("PWR OFF".equals(query) && ":".equals(response)) {
+            // When PWR OFF command is sent, next command can be sent 10 seconds after the colon is received
+            logger.debug("Refusing further commands for 10 seconds to power OFF completion");
+            ready = false;
+            ScheduledExecutorService scheduler = this.scheduler;
+            if (scheduler != null) {
+                scheduler.scheduleWithFixedDelay(() -> {
+                    ready = true;
+                }, 0, 10000, TimeUnit.MILLISECONDS);
```

Comment:
You need to keep track of the Future this returns so you can cancel it during disconnect.

Reply:
I think this was put in to get rid of a 10 second thread.sleep(). It should only be a one time delay for 10 seconds before setting ready back to true, so I changed it to schedule() instead.

Reply:
You still need to track the future so you can cancel it if the handler is disposed before 10 seconds expires.

Reply:
I think there's a glitch in the Matrix https://github.com/openhab/openhab-addons/pull/9021#discussion_r527972188

Reply:
I made it cancel on disconnect.